services/data-info/src/data_info/routes/avus.clj

```diff
@@ -12,26 +13,29 @@
 
     (GET* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :return AVUGetResult
+      :return MetadataListing
+      :middlewares [wrap-metadata-base-url]
       :summary "List AVUs (administrative)"
-      :description (str "List iRODS AVUs associated with a data item. Include administrative/system AVUs."
-(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE"))
+      :description
+          (str "List all AVUs associated with a data item. Include administrative/system iRODS AVUs."
+(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE")
+(get-endpoint-delegate-block "metadata" "GET /avus/{target-type}/{target-id}"))
       (svc/trap uri meta/admin-metadata-get data-id))
 
-    (POST* "/avus" [:as {uri :uri}]
+    (PUT* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :body [{:keys [irods-avus]} (describe AVUListing "An list of AVUs to add")]
+      :body [{:keys [irods-avus]} (describe AVUListing "A list of AVUs to add")]
       :return AVUChangeResult
       :summary "Add AVUs (administrative)"
-      :description (str "Associate AVUs with a data item. Allow adding any AVU."
+      :description (str "Associate iRODS AVUs with a data item. Allow adding any AVU."
 (get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE, ERR_DOES_NOT_EXIST, ERR_NOT_WRITEABLE, ERR_NOT_AUTHORIZED"))
       (svc/trap uri meta/admin-metadata-add data-id irods-avus))
```

Comment:
It was mentioned in PR #171 that if the client wants to update a metadata AVU, and preserve its `created_*` fields, then it needs to reuse the AVU's ID in the set request. So while we don't have endpoints that address a specific AVU by ID (other than the unused DELETE endpoint), the AVU IDs still have meaning in the set and update endpoints.

> or we should change the implementation to preserve IDs even when they aren't provided

It was addressed in PR #171 why it's not practical to handle this in the `set` endpoint, since it also handles batch AVU deletes in the same request.

> Plus, we don't support deletes with this sort of endpoint so it's not a complete patch.

Is this a requirement? After reviewing our /apps endpoints, it looks like we were already using `POST /apps` to create an app, `PUT /apps/:id` to update/replace a private app (which handles deleting app groups and properties like this metadata `set` endpoint), and `PATCH /apps` to only add/update fields given in the request and leave other fields alone.

I think I'm still leaning towards changing these metadata endpoints to follow the same pattern and use at least `PATCH` for the add/update endpoint. I have no strong opinion against using `PUT` for the `set` endpoint, and that would also be consistent with your arguments and what we're already using in the /apps endpoints.